tests/functional/test_user_class_with_slots.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import unicode_literals
+import pytest  # noqa
+from textx import metamodel_from_str
+
+grammar = """
+Shape:
+    'shape'
+    points+=Point
+    'end'
+;
+Point:
+    'point' x=INT y=INT
+;
+"""
+
+
+def test_user_class_with_slots():
+    """
+    User supplied meta class.
+    """
+    class Point(object):
+        "User class."
+        __slots__ = ['parent', 'x', 'y']
+
+        def __init__(self, parent, x, y):
+            self.parent = parent
+            self.x = x
+            self.y = y
+
+    modelstr = """
+    shape
+    point 34 45
+    point 12 23
+    end
+    """
+
+    mm = metamodel_from_str(grammar, classes=[Point], auto_init_attributes=False)
```

Comment:
Tiny fix is needed here for Travis tests to pass. We are using a max line length of 79 (PEP-8).